sklearn/decomposition/tests/test_nmf.py

```diff
@@ -933,3 +933,29 @@ def test_NMF_inverse_transform_W_deprecation():
 
     with pytest.warns(FutureWarning, match="Input argument `W` was renamed to `Xt`"):
         est.inverse_transform(W=Xt)
+
+
+def test_NMF_n_components_auto():
```

Comment:
Please write similar tests for NMF and MiniBatchNMF

Reply:
done in af565549c1339ed2fd3061c0b5b4532af2682b46
https://github.com/scikit-learn/scikit-learn/blob/af565549c1339ed2fd3061c0b5b4532af2682b46/sklearn/decomposition/tests/test_nmf.py#L938-L951